kotest-assertions/src/commonMain/kotlin/io/kotest/matchers/should.kt

```diff
@@ -1,26 +1,19 @@
 package io.kotest.matchers
 
-import io.kotest.assertions.AssertionCounter
-import io.kotest.assertions.ErrorCollector
-import io.kotest.assertions.Failures
-import io.kotest.assertions.clueContextAsString
-import io.kotest.assertions.collectOrThrow
-import io.kotest.assertions.compare
-import io.kotest.assertions.diffLargeString
-import io.kotest.assertions.stringRepr
-import io.kotest.mpp.sysprop
+import io.kotest.assertions.*
 
 @Suppress("UNCHECKED_CAST")
 infix fun <T, U : T> T.shouldBe(any: U?) {
    when (any) {
       is Matcher<*> -> should(any as Matcher<T>)
       else -> {
          AssertionCounter.inc()
-         if (this == null && any != null) {
-            ErrorCollector.collectOrThrow(equalsError(any, this))
-         } else if (!compare(this, any)) {
-            ErrorCollector.collectOrThrow(equalsError(any, this))
+         val throwable = when (this) {
```

Comment:
I think the first case in the when should be the check if one is null and one isn't (as they definitely can't be equal).

Reply:
Pull checking of actual value is null but expected value is not null in parent